File: backend/app/schemas/audit.py
"""Audit log read schema (M10)."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None
    action: str
    entity_type: str
    entity_id: int
    payload_json: str | None
    ip_address: str | None
    created_at: datetime

File: frontend/src/types/audit.ts
export interface AuditLog {
  id: number
  user_id: number | null
  action: string
  entity_type: string
  entity_id: number
  /** Raw JSON snapshot of the change, if recorded. */
  payload_json: string | null
  ip_address: string | null
  created_at: string
}
